### DISCUSSION TITLE
JuiceFS in a sidecar

### DISCUSSION THREAD

**Discussion by yzcwf** [juicedata/juicefs#4571] [General]
i want to run JuiceFS in a sidecar container to provide data access to other containers，but I faced the following questions：

If run JuiceFS on the host and mount a host path to a container, i can access the data in the container. However, if i mount the host path to the container and run JuiceFS in the container,  not be able to access the data from either the host or other containers.

**Comment by davies** [comment]
The CSI driver already support sidecar mode, have you tried that?

**Comment by yzcwf** [comment]
https://juicefs.com/docs/csi/introduction/
_____________________________________________________________________________________________________
Taking mount pod mode for example, this is the overall process:
User creates a PVC using an existing JuiceFS StorageClass;
PV is created and provisioned by CSI Controller, by default, a sub-directory named with PV ID will be created under JuiceFS root, settings controlling this process are defined (or referenced) in StorageClass definition.
Kubernetes PV Controller binds user-created PVC with the PV created by CSI Controller, PVC and PV both enter "Bound" state;
User creates application pod, referencing PVC previously created;
CSI Node Service creates mount pod on the associating node;
A JuiceFS Client runs inside the mount pod, and mounts JuiceFS volume to host, path being /var/lib/juicefs/volume/[pv-name];
CSI Node Service waits until mount pod is up and running, and binds PV with the associated container, the PV sub-directory is mounted in pod, path defined by volumeMounts;
Application pod is started by Kubelet.
___________________________________________________________________________________________________________
"A JuiceFS Client runs inside the mount pod, and mounts the JuiceFS volume to the host, with the path being /var/lib/juicefs/volume/[pv-name]." I referenced this implementation and ran the JuiceFS client in a Docker container. The mount path is mounted to the container through the host path. Everything is okay in the container, but I can't see the JuiceFS files on the host, so I'm not sure where the problem lies.

**Comment by davies** [comment]
>>> The mount path is mounted to the container through the host path

I think the bind-propagation should be configured:  https://docs.docker.com/storage/bind-mounts/#configure-bind-propagation

**Reply by yzcwf** [reply]
it was this configure that caused the problem. Thank you very much.
